help/en/html/doc/Technical/Patterns.shtml

```diff
@@ -33,6 +33,38 @@ structure and patterns by looking at older code pieces.
 This page attempts to describe the recommended structure and patterns,
 and point to examples of current best practices.
 
+<a id="namedbeans" name="namedbeans"></a><h2>Names, NamedBeans, and Managers</h2>
```

Comment:
Thanks!

The ConfigureXML systems allows storing and retrieving parameters with Object keys, so long as they can be recreated from a String ctor. 

One implementation is at line 210 in java/src/jmri/managers/configurexml/AbstractNamedBeanManagerConfigXML.java

That loads a <key> element then creates an object of the type in the class="" attribute.  The only uses I found are Strings, i.e.:

````
java/test//jmri/configurexml/loadref/BlockAndSignalMastTest.xml:          <key class="java.lang.String">intermediateSignal</key>
````

So I think we could switch the property key to String unless there's some use of this somewhere else.